homeassistant/components/apcupsd/binary_sensor.py

```diff
@@ -34,8 +34,8 @@ def name(self):
     @property
     def is_on(self):
         """Return true if the UPS is online, else false."""
-        return self._state == apcupsd.VALUE_ONLINE
+        return self._state & apcupsd.VALUE_ONLINE > 0
 
     def update(self):
         """Get the status report from APCUPSd and set this entity's state."""
-        self._state = self._data.status[apcupsd.KEY_STATUS]
+        self._state = int(self._data.status[apcupsd.KEY_STATUS], 16)
```

Comment:
Does this work? `KEY_STATUS` used to contain the word `ONLINE` but now you cast it to an int ?

Reply:
I updated `KEY_STATUS` to point to the `STATFLAG` field from apcupsd instead of the string status field. The `STATFLAG` comes through as a string but is hex, e.g. `0x05000008`, where the [3rd bit represents online](http://www.apcupsd.org/manual/manual.html#status-bits).

Reply:
Sorry, missed that part.